### ISSUE TITLE
Uncaught TypeError: Cannot read property 'class' of null

### ISSUE THREAD

**Issue by openspectrum** [atom/find-and-replace#348]
[Enter steps to reproduce below:]

1. Open Markdown file and a Markdown preview
2. Cmd-F to search, start typing text (options: case insensitive, no regex, whole document)
3. Hit enter => error message, even though search works

**Atom Version**: 0.178.0
**System**: Mac OS X 10.10.2
**Thrown From**: [find-and-replace](https://github.com/atom/find-and-replace) package, v0.156.0

### Stack Trace

Uncaught TypeError: Cannot read property 'class' of null

```
At /Applications/Atom.app/Contents/Resources/app/src/text-editor-presenter.js:953

TypeError: Cannot read property 'class' of null
  at TextEditorPresenter.module.exports.TextEditorPresenter.highlightDidFlash (/Applications/Atom.app/Contents/Resources/app/src/text-editor-presenter.js:953:43)
  at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:82:11)
  at Decoration.module.exports.Decoration.flash (/Applications/Atom.app/Contents/Resources/app/src/decoration.js:171:27)
  at /Applications/Atom.app/Contents/Resources/app/src/selection.js:133:37
  at Selection.module.exports.Selection.modifySelection (/Applications/Atom.app/Contents/Resources/app/src/selection.js:912:7)
  at Selection.module.exports.Selection.setBufferRange (/Applications/Atom.app/Contents/Resources/app/src/selection.js:118:19)
  at /Applications/Atom.app/Contents/Resources/app/src/text-editor.js:1787:43
  at TextEditor.module.exports.TextEditor.mergeIntersectingSelections (/Applications/Atom.app/Contents/Resources/app/src/text-editor.js:2131:43)
  at TextEditor.module.exports.TextEditor.setSelectedBufferRanges (/Applications/Atom.app/Contents/Resources/app/src/text-editor.js:1779:19)
  at TextEditor.module.exports.TextEditor.setSelectedBufferRange (/Applications/Atom.app/Contents/Resources/app/src/text-editor.js:1762:19)
  at FindView.module.exports.FindView.selectMarkerAtIndex (/Applications/Atom.app/Contents/Resources/app/node_modules/find-and-replace/lib/find-view.js:623:16)
  at FindView.module.exports.FindView.selectFirstMarkerAfterCursor (/Applications/Atom.app/Contents/Resources/app/node_modules/find-and-replace/lib/find-view.js:538:19)
  at /Applications/Atom.app/Contents/Resources/app/node_modules/find-and-replace/lib/find-view.js:3:61
  at FindView.module.exports.FindView.findAndSelectResult (/Applications/Atom.app/Contents/Resources/app/node_modules/find-and-replace/lib/find-view.js:430:9)
  at FindView.findAndSelectResult (/Applications/Atom.app/Contents/Resources/app/node_modules/find-and-replace/lib/find-view.js:3:61)
  at FindView.module.exports.FindView.findNext (/Applications/Atom.app/Contents/Resources/app/node_modules/find-and-replace/lib/find-view.js:407:19)
  at FindView.findNext (/Applications/Atom.app/Contents/Resources/app/node_modules/find-and-replace/lib/find-view.js:3:61)
  at FindView.module.exports.FindView.confirm (/Applications/Atom.app/Contents/Resources/app/node_modules/find-and-replace/lib/find-view.js:373:19)
  at atom-text-editor.subscriptions.add.atom.commands.add.core:confirm (/Applications/Atom.app/Contents/Resources/app/node_modules/find-and-replace/lib/find-view.js:240:26)
  at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app/src/command-registry.js:243:29)
  at /Applications/Atom.app/Contents/Resources/app/src/command-registry.js:3:61
  at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/Applications/Atom.app/Contents/Resources/app/node_modules/atom-keymap/lib/keymap-manager.js:558:16)
  at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/Applications/Atom.app/Contents/Resources/app/node_modules/atom-keymap/lib/keymap-manager.js:396:22)
  at HTMLDocument.module.exports.WindowEventHandler.onKeydown (/Applications/Atom.app/Contents/Resources/app/src/window-event-handler.js:172:20)

```

### Commands

```
     -0:35.0 editor:move-to-end-of-screen-line (atom-text-editor.editor)
  2x -0:34.8 editor:newline (atom-text-editor.editor)
     -0:34.5 core:paste (atom-text-editor.editor)
     -0:33.9 core:move-to-top (atom-text-editor.editor)
     -0:33.7 core:move-up (atom-text-editor.editor)
     -0:32.7 find-and-replace:show (atom-text-editor.editor)
  2x -0:31.7 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:26.1 editor:consolidate-selections (atom-text-editor.editor.mini.is-focused)
     -0:26.1 core:cancel (atom-text-editor.editor.mini.is-focused)
     -0:06.9 window:focus-pane-on-left (atom-text-editor.editor)
     -0:04.0 find-and-replace:show (atom-text-editor.editor)
     -0:02.6 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:01.4 editor:consolidate-selections (atom-text-editor.editor.mini.is-focused)
     -0:01.4 core:cancel (atom-text-editor.editor.mini.is-focused)
     -0:00.8 find-and-replace:show (atom-text-editor.editor)
     -0:00.6 core:confirm (atom-text-editor.editor.mini.is-focused)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "autoclose-html",
      "language-java",
      "language-coffee-script",
      "language-clojure",
      "language-less",
      "language-mustache",
      "language-perl",
      "language-php",
      "language-python",
      "language-toml",
      "linter",
      "linter-rubocop"
    ],
    "themes": [
      "atom-dark-ui",
      "solarized-light-syntax"
    ]
  }
}
```

### Installed Packages

```coffee
# User
alignment, v1.1.4
atom-beautify, v0.21.3
change-case, v0.4.0

# Dev
No dev packages
```

**Comment by izuzak** [comment]
@openspectrum Having trouble reproducing this. Is it happening in [safe mode](https://github.com/atom/atom/blob/master/docs/debugging.md#check-if-the-problem-shows-up-in-safe-mode) as well? For every Markdown file or only some?

**Comment by izuzak** [comment]
Ahh, yeah, I can reproduce with a string which is in the document, was trying with a string which wasn't in the document. :sweat_smile:

**Comment by openspectrum** [comment]
Aha, yes. Sorry for the omission of that detail.

**Comment by izuzak** [comment]
@nathansobo Since the exception was thrown from TextEditorPresenter, wondering if you see what's causing problems here? (I'm able to reproduce on Atom 0.179.0-090561e)

**Comment by nathansobo** [comment]
Thanks guys. Looks like I forgot to destroy the `TextEditorPresenter` when the editor component is unmounted :see_no_evil:. Running specs now and should have a fix out shortly.

**Comment by openspectrum** [comment]
Great, thanks! Glad it wasn't anything too difficult. :D

**Comment by zdne** [comment]
:+1: just ran into this as well

**Comment by nathansobo** [comment]
We'll be releasing a fix later today.

**Comment by crucialfelix** [comment]
I love that the error message linked me to this issue, I was able to visit and read about the fix and now I've updated and got the fix !  (0.179.0 2/11/2015, 9:48:18 PM) 

That integration with the github issues is really nice work !

**Comment by nathansobo** [comment]
@crucialfelix Thanks very much. I guess it sweetens the bitterness of having the issue in the first place a bit.
